src/compile/data/bin.ts

```diff
@@ -89,6 +90,27 @@ export class BinNode extends DataFlowNode {
     return new BinNode(bins);
   }
 
+  public static makeBinFromTransform(model: Model, t: BinTransform) {
+    const bins: Dict<BinComponent> = {};
+
+    if (t.bin) {
+      const bin: Bin = isBoolean(t.bin) ? {} : t.bin;
+      const key = `${binToString(t.bin)}_${t.field}`;
```

Comment:
We need to augment `bin` with `maxbins` if both `maxbins` and step` are` not provided like in `normalize` in fielddef.ts